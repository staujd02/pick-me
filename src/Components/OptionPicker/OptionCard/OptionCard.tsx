import React from 'react';
import Typography from '@material-ui/core/Typography';
import CardActionArea from '@material-ui/core/CardActionArea';
import Card from '@material-ui/core/Card';

const OptionCard: React.FunctionComponent<OptionCardProps> = ({handleClick, option}) => (
    <Card>
        <CardActionArea onClick={handleClick}>
            <Typography>{option.title}</Typography>
            <img src={option.image} />
        </CardActionArea>
    </Card>
)

export default OptionCard;