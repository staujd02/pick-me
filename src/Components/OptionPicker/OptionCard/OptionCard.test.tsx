import React from 'react';
import { shallow, ShallowWrapper } from "enzyme";
import { formatHTML } from '../../../TestUtilities/htmlFormatter';
import OptionCard from './OptionCard';

describe('The Option Picker', () => {

    let wrapper: ShallowWrapper<OptionCardProps>;
    let props: OptionCardProps;

    let handleClick: jest.Mock;

    beforeEach(() => {
        props = {
            handleClick: (handleClick = jest.fn()),
            option: {
                image: "image",
                title: "title"
            }
        }
        wrapper = shallow(<OptionCard {...props} />);
    });

    it('renders correctly', () => expect(formatHTML(wrapper.html())).toMatchSnapshot());

    describe('when the option is displayed', () => {
        it('shows the user the title of that option', () => {
            let firstRenderedOptionTitle = getFirst(wrapper, typographyIdentifier);
            expect(
                firstRenderedOptionTitle.contains(props.option.title)
            ).toBeTruthy();
        });

        it('shows the user that options image', () => {
            let firstRenderedOptionImg = getFirst(wrapper, imageIdentifier);
            expect(firstRenderedOptionImg.prop('src')).toEqual(props.option.image);
        });
    });

    describe('when the user clicks on the card action area', () => {

        beforeEach(() => {
            wrapper.find(cardActionAreaIdentifier).simulate('click');
        });

        it('calls the registered onClick handler', () => {
            expect(handleClick).toHaveBeenCalled();
        });
    });

    function getFirst(wrapper: ShallowWrapper<OptionCardProps>, identifier: string) {
        return wrapper.find(identifier).first();
    }

    const typographyIdentifier = 'WithStyles(ForwardRef(Typography))';
    const cardActionAreaIdentifier = 'WithStyles(ForwardRef(CardActionArea))';
    const imageIdentifier = 'img';
});