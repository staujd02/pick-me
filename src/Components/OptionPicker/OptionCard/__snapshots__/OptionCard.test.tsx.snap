// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Option Picker renders correctly 1`] = `
"
<div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded\\">
  <button class=\\"MuiButtonBase-root MuiCardActionArea-root\\" tabindex=\\"0\\" type=\\"button\\">
    <p class=\\"MuiTypography-root MuiTypography-body1\\">title</p>
    <img src=\\"image\\">
    <span class=\\"MuiCardActionArea-focusHighlight\\"></span>
  </button>
</div>
"
`;
